windows/RNCNetInfoCPP/RNCNetInfo.cpp

```diff
@@ -138,19 +152,23 @@ namespace winrt::ReactNativeNetInfo::implementation {
                 auto connectivityLevel = profile.GetNetworkConnectivityLevel();
                 auto signal = profile.GetSignalBars();
                 auto costType = profile.GetConnectionCost().NetworkCostType();
+                auto isWifiConnection = profile.IsWlanConnectionProfile() || requestedInterface.find("wifi") != std::string::npos;
+                auto isCellularConnection = profile.IsWwanConnectionProfile() || requestedInterface.find("cellular") != std::string::npos;
+                auto isEthernetConnection = networkAdapter || requestedInterface.find("ethernet") != std::string::npos;
+                auto isConnectionExpensive = costType == NetworkCostType::Fixed || costType == NetworkCostType::Variable;
 
                 state.isConnected = connectivityLevel != NetworkConnectivityLevel::None;
 
+                NetInfoDetails details{};
                 if (state.isConnected) {
-                    NetInfoDetails details{};
-
-                    state.isInternetReachable = connectivityLevel == NetworkConnectivityLevel::InternetAccess;
-                    details.isConnectionExpensive = costType == NetworkCostType::Fixed || costType == NetworkCostType::Variable;
                     if (signal) {
                         details.strength = winrt::unbox_value<uint8_t>(signal) * 20; // Signal strength is 0-5 but we want 0-100.
                     }
+                    if (isWifiConnection) {
+                        if (!profile.IsWlanConnectionProfile()) {
+                            throw ("Wifi profile is not available");
+                        }

```

Comment:
The error will be caught by the catch down below, and then we'll return a partially filled out NetInfoState that may indicate the requested interface is connected even though that interface doesn't exist.

The catch should probably reset NetInfoState to an empty state.

Reply:
Thanks for calling that out @aschultz, I've reset the state on error.